src/Streamly/Internal/Prelude.hs

```diff
@@ -4192,3 +4196,16 @@ usingStateT s f xs = evalStateT s $ f $ liftInner xs
 {-# INLINE runStateT #-}
 runStateT :: Monad m => s -> SerialT (StateT s m) a -> SerialT m (s, a)
 runStateT s xs = fromStreamD $ D.runStateT s (toStreamD xs)
+
+------------------------------------------------------------------------------
+-- Time related
+------------------------------------------------------------------------------
+
+{-# INLINE takeByTime #-}
+takeByTime :: (IsStream t, MonadIO m) => Double -> t m a -> t m a
```

Comment:
Some documentation would be nice. We need to mention that the time check is done before generating the stream element, therefore, the total actual time may differ depending on how long it takes to generate the stream element.